dipy/denoise/noise_estimate.py

```diff
@@ -203,15 +188,21 @@ def piesno_3D(data, N, alpha=0.01, l=100, itermax=100, eps=1e-5):
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give up
+    if m == 0:
```

Comment:
Well, the first estimate is based on a quantile, starting from ~0.79 to the median (0.5). If the first estimate by the quantile is 0, then at least more than half of the voxels on the current slice are 0. In this case the function simply exit with a warning.

That being said, the computed thresholds afterward are going ot be really low, but could still find stuff. I could change that for a sclice to be exactly zero before exiting, and worse case scenario no voxel should be found anyway.